synapse/lib/iq.py

```diff
@@ -42,6 +44,31 @@
 
 logger = logging.getLogger(__name__)
 
+def objhierarchy(obj):
+    '''
+    Return the type hierarchy of an an object.
+    Dictionary objects have their key values preserved.
+
+    This function exists for debugging purposes.
+    '''
+    # Return fast on generators
+    if isinstance(obj, types.GeneratorType):
+        return type(obj)
+    # Known objects we want to return fast on
+    if isinstance(obj, (str, int, float, bytes)):
+        return type(obj)
+    # Iterables we care about
+    if isinstance(obj, collections.Iterable):
```

Comment:
We could probably check if the object is an `Iterable` first, then juts return `type(obj)` if it is not.